src/osp/Active/SysVehicle.cpp

```diff
@@ -317,40 +313,73 @@ ActiveEnt SysVehicle::part_instantiate(PrototypePart& part,
             shader.m_lightPosition = Vector3{10.0f, 15.0f, 5.0f};
             shader.m_ambientColor = 0x111111_rgbf;
             shader.m_specularColor = 0x330000_rgbf;
-            m_scene.reg_emplace<Phong::ACompPhongInstance>(currentEnt, std::move(shader));
+            rScene.reg_emplace<Phong::ACompPhongInstance>(currentEnt, std::move(shader));
 
-            CompDrawableDebug& bBocks = m_scene.reg_emplace<CompDrawableDebug>(
+            CompDrawableDebug& bBocks = rScene.reg_emplace<CompDrawableDebug>(
                         currentEnt, meshRes, &Phong::draw_entity, 0x0202EE_rgbf);
 
             //new DrawablePhongColored(*obj, *m_shader, *mesh, 0xff0000_rgbf, m_drawables);
         }
         else if (currentPrototype.m_type == ObjectType::COLLIDER)
         {
-            ACompCollisionShape& collision = m_scene.reg_emplace<ACompCollisionShape>(currentEnt);
+            ACompCollisionShape& collision = rScene.reg_emplace<ACompCollisionShape>(currentEnt);
             const ColliderData& cd = std::get<ColliderData>(currentPrototype.m_objectData);
             collision.m_shape = cd.m_type;
 
         }
     }
 
-
-    // first element is 100% going to be the root object
-
-    // Temporary: add a rigid body root
-    //CompNewtonBody& nwtBody
-    //        = m_scene->get_registry().emplace<CompNewtonBody>(newEntities[0]);
-
     // Create mass
-    m_scene.reg_emplace<ACompMass>(newEntities[0], part.get_mass());
+    rScene.reg_emplace<ACompMass>(newEntities[0], part.get_mass());
 
     // return root object
     return newEntities[0];
 }
 
+void SysVehicle::update_activate(ActiveScene &rScene)
+{
+    ACompAreaLink *pArea = SysAreaAssociate::try_get_area_link(rScene);
+
+    if (pArea == nullptr)
+    {
+        return;
+    }
+
+    Universe &rUni = pArea->get_universe();
+    TypeSatIndex type = rUni.sat_type_find_index<universe::SatVehicle>();
+    ActivationTracker& activations = pArea->get_tracker(type);
+
+    // Delete vehicles that have gone too far from the ActiveArea range
+    for (auto const &[sat, ent] : activations.m_leave)
+    {
+        rScene.hier_destroy(ent);
```

Comment:
(void)sat; before rScene.hier_destroy(ent) to shut up the compiler about unused variable warnings.